Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanViewModel.kt

```diff
@@ -6,29 +6,39 @@ import com.squareup.inject.assisted.AssistedInject
 import de.rki.coronawarnapp.exception.ExceptionCategory
 import de.rki.coronawarnapp.exception.TransactionException
 import de.rki.coronawarnapp.exception.http.CwaWebException
+import de.rki.coronawarnapp.exception.reporting.ErrorCodes
+import de.rki.coronawarnapp.exception.reporting.ReportedIOException
 import de.rki.coronawarnapp.exception.reporting.report
 import de.rki.coronawarnapp.service.submission.QRScanResult
 import de.rki.coronawarnapp.service.submission.SubmissionService
+import de.rki.coronawarnapp.storage.LocalData
 import de.rki.coronawarnapp.ui.submission.ApiRequestState
 import de.rki.coronawarnapp.ui.submission.ScanStatus
 import de.rki.coronawarnapp.ui.submission.viewmodel.SubmissionNavigationEvents
 import de.rki.coronawarnapp.util.Event
+import de.rki.coronawarnapp.util.formatter.TestResult
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import timber.log.Timber
 
-class SubmissionQRCodeScanViewModel @AssistedInject constructor() : CWAViewModel() {
-
+class SubmissionQRCodeScanViewModel @AssistedInject constructor() :
+    CWAViewModel() {
     val routeToScreen: SingleLiveEvent<SubmissionNavigationEvents> = SingleLiveEvent()
     private val _scanStatus = MutableLiveData(Event(ScanStatus.STARTED))
 
     val scanStatus: LiveData<Event<ScanStatus>> = _scanStatus
+    val showRedeemedTokenWarning = SingleLiveEvent<Unit>()
+
+    open class InvalidQRCodeException : ReportedIOException(
+        ErrorCodes.CWA_WEB_REQUEST_PROBLEM.code, "error in qr code"
```

Comment:
I don't think this is a WEB_REQUEST_PROBLEM, you could declare it to simply extend Exception as it is only used internally

Reply:
@kolyaopahle  How do you feel about a generic Exception here?

  open class InvalidQRCodeException : Exception("error in qr code" )

Reply:
Used the about mentioned implmentation. check it out when you can.